lib/course_creation_manager.rb

```diff
@@ -115,4 +120,39 @@ def add_instructor_to_course
   def add_tags_to_course
     TagManager.new(@course).initial_tags(creator: @instructor)
   end
+
+  def create_all_category(categories, depth, source)
+    categories.each do |category|
+      name = ArticleUtils.format_article_title(category[:value])
+      category = Category.find_or_create_by(name:, depth:, wiki: @wiki, source:)
+      @overrides[:categories] << category
+    end
+  end
+
+  def set_scoping_methods
+    return if @scoping_methods.nil? || @scoping_methods.empty?
+    @overrides[:categories] = []
+
+    add_categories_to_course @scoping_methods[:categories] if @scoping_methods[:categories]
+    add_templates_to_course @scoping_methods[:templates] if @scoping_methods[:templates]
+    add_page_pile_to_course @scoping_methods[:pagepile] if @scoping_methods[:pagepile]
+    add_petscan_to_course @scoping_methods[:petscan] if @scoping_methods[:petscan]
+  end
+
+  def add_categories_to_course(category_params)
+    depth = category_params[:depth]
+    create_all_category(category_params[:tracked], depth, 'category')
```

Comment:
For each of this `add_X_to_course` methods, is it possible for the sub-params like `category_params[:tracked]` to be `nil` or to throw a NoMethodError for `[ ]`, even though the param itself made it through the guard statement of `@scoping_methods[:categories]`?